Mage/src/main/java/mage/abilities/effects/common/ReturnSourceFromGraveyardToBattlefieldEffect.java

```diff
@@ -38,10 +39,15 @@ public ReturnSourceFromGraveyardToBattlefieldEffect(boolean tapped, boolean owne
     }
 
     public ReturnSourceFromGraveyardToBattlefieldEffect(boolean tapped, boolean ownerControl, boolean haste) {
+        this(tapped, ownerControl, haste, false);
+    }
+
+    public ReturnSourceFromGraveyardToBattlefieldEffect(boolean tapped, boolean ownerControl, boolean haste, boolean attacking) {
         super(Outcome.PutCreatureInPlay);
         this.tapped = tapped;
         this.ownerControl = ownerControl;
         this.haste = haste;
+        this.attacking = attacking;
         setText();
```

Comment:
One last thing, should have mentioned it before. But can you please change the way setText works so that it returns a String so that staticText is set in the constructor (for readability) rather than inside `setText`.

Reply:
Sure, done.